### PR TITLE
Updated cross function

### LINKED ISSUE #5531: cross function produces null when there is no match
I notice that the result of the `cross()` function is not an array when there is no match found - instead, the result is `null`. This seems to contradict the reference documentation which states "Returns an array of **zero** or more rows..." (that is, we should expect an an empty array, `[]`, when there are no matching rows). So if, for instance, you are wanting a count of the number of matches using `cross(...).length()` (which is something I do) then you currently get "Error: length expects an array or a string" where you expect to get 0.

Incidentally, if the function gets fixed to produce an empty array, as per the documentation, then the result of a common operation where you just want to retrieve the first matching row (i.e. `cross(...)[0]`) will, when there is no match, change to "Error: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0" - currently the result is just null (since `null[0]` is also null). So I'm wondering if the null return was implemented as a specific exception to avoid this ungainly instance. Should we consider making the result of indexing an empty array (e.g. `[][0]`) null instead of producing the error?

I'm looking at OpenRefine 3.6.2 [579a6f7].

### PR DESCRIPTION
**Pull request by jenny-Musah**
Fixes #5531

Changes proposed in this pull request:
- made the cross function return an empty array if there is no matching rows

### CODE REVIEW

**Comment by jenny-Musah** [comment]
@wetneb 

I made another pull request, Thanks, I hope this one passes

**Comment by jenny-Musah** [comment]
user@DESKTOP-VCH2N0O MINGW64 ~/IdeaProjects/OpenRefine (master)
$ mvn formatter:format
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 12 modules...
[INFO] Installing Nexus Staging features:                                                          
[INFO]   ... total of 12 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------                    
[INFO] Reactor Build Order:                                                                        
[INFO]                                                                                             
[INFO] OpenRefine                                                         [pom]
[INFO] OpenRefine - main                                                  [jar]
[INFO] OpenRefine - server                                                [jar]
[INFO] OpenRefine - extensions                                            [pom]
[INFO] OpenRefine - Jython extension                                      [jar]
[INFO] OpenRefine - Wikibase extension                                    [jar]
[INFO] OpenRefine - Database extension                                    [jar]
[INFO] OpenRefine - Gdata extension                                       [jar]
[INFO] OpenRefine - PC-axis extension                                     [jar]
[INFO] OpenRefine - Phonetic clustering extension                         [jar]
[INFO] OpenRefine - packaging                                             [pom]
[INFO] OpenRefine Java JMH benchmarks                                     [jar]
[INFO] 
[INFO] ---------------------< org.openrefine:openrefine >----------------------
[INFO] Building OpenRefine 3.8-SNAPSHOT                                  [1/12]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- formatter-maven-plugin:2.17.1:format (default-cli) @ openrefine ---
[INFO] 
[INFO] ------------------------< org.openrefine:main >-------------------------
[INFO] Building OpenRefine - main 3.8-SNAPSHOT                           [2/12]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- formatter-maven-plugin:2.17.1:format (default-cli) @ main ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for OpenRefine 3.8-SNAPSHOT:
[INFO]
[INFO] OpenRefine ......................................... SUCCESS [  3.516 s]
[INFO] OpenRefine - main .................................. FAILURE [  0.409 s]
[INFO] OpenRefine - server ................................ SKIPPED
[INFO] OpenRefine - extensions ............................ SKIPPED
[INFO] OpenRefine - Jython extension ...................... SKIPPED
[INFO] OpenRefine - Wikibase extension .................... SKIPPED
[INFO] OpenRefine - Database extension .................... SKIPPED
[INFO] OpenRefine - Gdata extension ....................... SKIPPED
[INFO] OpenRefine - PC-axis extension ..................... SKIPPED
[INFO] OpenRefine - Phonetic clustering extension ......... SKIPPED
[INFO] OpenRefine - packaging ............................. SKIPPED
[INFO] OpenRefine Java JMH benchmarks ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.604 s
[INFO] Finished at: 2023-03-21T21:58:30+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.revelc.code.formatter:formatter-maven-plugin:2.17.1:format (default-cli) on project main: Cannot find config file [C:/Users/user/IDEs/eclipse/Refine.style.xml] -> [Help 1]   
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :main

**Comment by jenny-Musah** [comment]
That is the error i got

**Comment by jenny-Musah** [comment]
Nope, the file does not exist in that location.
When I checked the version this is what i got
C:\Users\user>mvn -v
Apache Maven 3.9.1 (2e178502fcdbffc201671fb2537d0cb4b4cc58f8)
Maven home: C:\Program Files\Maven\apache-maven-3.9.1
Java version: 18.0.1.1, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk-18.0.1.1
Default locale: en_GB, platform encoding: UTF-8
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"


Thank you so much, it passed already, so if I pull from this branch , it should run fine, right?

**Comment by jenny-Musah** [comment]
@wetneb 
The second commit for the test, the checks are still yet pass, what do I do?

**Comment by wetneb** [comment]
No it should not be needed, the code is already formatted correctly.

File: main/tests/server/src/com/google/refine/expr/functions/CrossTests.java [hunk 1]
```diff
@@ -64,6 +64,8 @@ public void init() {
         logger = LoggerFactory.getLogger(this.getClass());
     }
 
+    private HasFieldsListImpl rows;
```
Comment (wetneb) [inline main/tests/server/src/com/google/refine/expr/functions/CrossTests.java:67]:
```suggestion
    private HasFieldsListImpl emptyList;
```
I would rather call this `emptyList` to make it clearer that it is used for test cases where we expect the return value of the function to be empty. (Of course changing this attribute name will require updating the rest of the test.

Comment (jenny-Musah) [reply main/tests/server/src/com/google/refine/expr/functions/CrossTests.java:67]: @wetneb 
Okay, I would do that

**Review by wetneb — APPROVED** [review]
Thank you and congratulations for this PR!

**Pull request merged**